### DISCUSSION TITLE
Add group- support for tailwind CSS plugin variants (ui-*)

### DISCUSSION THREAD

**Discussion by peteruithoven** [tailwindlabs/tailwindcss#11527] [Ideas]
When using the neat variants from the tailwind CSS plugin for Headless UI, like `ui-selected`, `ui-not-selected` people can run into quicks when applying them to the children of the Headless UI elements. 

See for example the following example:
https://play.tailwindcss.com/6u0O6F5iGj
Copy of that code:
``` jsx
<!-- Simplified generated html from Headless UI Listbox -->
<ul data-headlessui-state="open">
  <li data-headlessui-state="">
    <span class="ui-not-selected:invisible">✓</span>
    default
  </li>

  <li data-headlessui-state="selected">
    <!-- doesn't work-->
    <span class="ui-not-selected:invisible">✓</span>
    selected
  </li>

  <li data-headlessui-state="selected">
    <!-- works-->
    <span class="invisible ui-selected:visible">✓</span>
    selected
  </li>
</ul>
```

Could these cases be simplified by adding support for these variants to the `group-*` variants?

Related to a prev. discussion: https://github.com/tailwindlabs/tailwindcss/discussions/11526

**Comment by Gilbertrockwood** [comment]
Hello,

In short, adding support for the ui-selected and ui-not-selected variants to the group-* variants in Tailwind CSS could simplify the cases where these variants are applied to children of Headless UI elements.  [MyAccountAccess Online](https://www.myaccountaccess.one/)

By enabling the ui-selected and ui-not-selected variants within the group-* variants, it would allow for more streamlined and concise code when applying these variants to the children of Headless UI elements. This enhancement would provide a more intuitive and consistent experience for developers working with Tailwind CSS and Headless UI, potentially reducing any quirks or challenges encountered in such scenarios.
